jdi-light/src/main/java/com/epam/jdi/light/elements/base/JdiSettings.java

```diff
@@ -68,10 +68,10 @@ private static WebElement getWithArgs(JDIBase b, Object[] args) {
 
     // region Utilities
     public static WebElement filterWebListToWebElement(JDIBase base, List<WebElement> els) {
```

Comment:
decisions becomes to complex, so better to change this code to something like this:
```
public static WebElement filterWebListToWebElement(JDIBase base, List<WebElement> els) {
	if (els.isEmpty())
		throw exception(FAILED_TO_FIND_ELEMENT_MESSAGE, base.toString(), base.getTimeout());
	SearchType searchType = base.searchType == DEFAULT ? SEARCH_TYPE : base.searchType;
	switch (searchType) {
		case(SOFT):
			return softSearch(base, els);
		case(FAST):
			return fastSearch(base, els);
		case(STRICT):
			return strictSearch(base, els);
		default:
			throw exception("Unknown search type");
	}
}
```